tools/sync/commands.py

```diff
@@ -67,35 +70,68 @@ def __init__(self, base):
         self.logger = logging.getLogger(__name__)
         logging.basicConfig()
 
+    def run_command(command):
+        cmd = Command(command,
+                      args_subst={"PROJECT": self.name},
+                      args_append=[self.name], excl_subst=True)
+        cmd.execute()
+        self.retcodes[str(cmd)] = cmd.getretcode()
+        self.outputs[str(cmd)] = cmd.getoutput()
+
     def run(self):
         """
         Run the sequence of commands and capture their output and return code.
         First command that returns code other than 0 terminates the sequence.
         If the command has return code 2, the sequence will be terminated
         however it will not be treated as error.
+
+        Any command entry that is a URI, will be used to submit RESTful API
+        request. Return codes for these requests are not checked.
         """
 
+        PROJECT_SUBST = '%PROJECT%'
         for command in self.commands:
-            cmd = Command(command,
-                          args_subst={"ARG": self.name},
-                          args_append=[self.name], excl_subst=True)
-            cmd.execute()
-            self.retcodes[str(cmd)] = cmd.getretcode()
-            self.outputs[str(cmd)] = cmd.getoutput()
-
-            # If a command fails, terminate the sequence of commands.
-            retcode = cmd.getretcode()
-            if retcode != 0:
-                if retcode == 2:
-                    self.logger.info("command '{}' requested break".
-                                     format(cmd))
-                    self.run_cleanup()
+            if is_web_uri(command[0]):
+                uri = command[0].replace(PROJECT_SUBST, self.name)
+                verb = command[1]
+                data = command[2]
+                if len(data) == 0:
+                    # PUT/POST need the data.
+                    if verb in ['PUT', 'POST']:
+                        self.logger.error('Empty data for PUT/POST')
```

Comment:
I think there are some cases when `PUT` and `POST` can have an empty body in our REST impl. E.g. marking project as indexed : `/api/v1/{project}/indexed`...

Maybe just log a warning? Or if you prefer you can change the REST API but I did not know how :/

Reply:
We can return something in the REST API - the updated state of the entity or some message in json? Not that important though.

Reply:
Aha, removed.